firedrake/mg/ufl_utils.py

```diff
@@ -138,11 +138,22 @@ def coarsen_function_space(V, self, coefficient_mapping=None):
 
     mesh = self(V.mesh(), self)
 
+    Vf = V
     V = firedrake.FunctionSpace(mesh, V.ufl_element())
+
+    from firedrake.dmhooks import get_transfer_operators, push_transfer_operators
+    transfer = get_transfer_operators(Vf.dm)
+    push_transfer_operators(V.dm, *transfer)
+    if len(V) > 1:
+        for V_, Vc_ in zip(Vf, V):
+            transfer = get_transfer_operators(V_.dm)
+            push_transfer_operators(Vc_.dm, *transfer)
```

Comment:
Here is another place where we coarsen function spaces (not via DMCoarsen) so we must attach transfer operators here. This should be refactored so that there is not code duplication with the similar stuff in dmhooks.py